src/resolver.rs

```diff
@@ -211,6 +211,13 @@ impl<'ctx> DependencyResolver<'ctx> {
 
         // Register the versions.
         for (name, id) in names {
+            if name == self.sess.manifest.package.name {
```

Comment:
```suggestion
            if name == manifest.package.name {
```
Since `manifest` is a function argument, that should be used instead of the `manifest` in `sess`, right? The `register_dependencies_in_manifest` function is not always called with `manifest == self.sess.manifest`.

Reply:
This is true, however the behavior is different, as in one case the top package name is compared with all dependencies called throughout the dependency tree, while in the other case it is only checked if a package does not depend on itself. I have added the second test with the new error case as well.

Reply:
I see, thanks for elaborating!